app/src/main/java/com/androiddevs/news/extension/Extensions.kt

```diff
@@ -0,0 +1,79 @@
+package com.androiddevs.news.extension
+
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
+import com.androiddevs.news.ui.LoginActivity
+import com.androiddevs.news.ui.NewsActivity
+import com.androiddevs.news.ui.RegistrationActivity
+
+fun Activity.displayToast(message: String) {
+    Toast.makeText(this.applicationContext, message, Toast.LENGTH_LONG).show()
+}
+
+fun Activity.displayToast(message: Int) {
+    Toast.makeText(this.applicationContext, message, Toast.LENGTH_LONG).show()
+}
+
+fun Fragment.displayToast(message: String) {
+    Toast.makeText(this.context, message, Toast.LENGTH_LONG).show()
+}
+
+fun ViewGroup.inflateView(layout: Int, attachToRoot: Boolean = false): View =
+    LayoutInflater.from(this.context).inflate(layout, this, attachToRoot)
+
+inline fun <T> LifecycleOwner.bind(
+    liveData: LiveData<T>,
+    crossinline onChanged: (T) -> Unit
+) {
+    liveData.observe(this) { onChanged.invoke(it) }
+}
+
+fun makeVisible(it: Boolean, progressBar: ProgressBar) {
+    if (it) {
+        progressBar.visibility = View.VISIBLE
+    } else {
+        progressBar.visibility = View.INVISIBLE
+    }
+}
+
+fun Activity.isPermissionGranted(permission: String) =
+    Build.VERSION.SDK_INT < 23 || checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED
+
+
+fun Activity.requestPermission(permission: String): Boolean {
+    ActivityCompat.requestPermissions(
+        this,
+        Array<String>(1) { permission },
+        1
+    )
+    return isPermissionGranted(permission)
+}
+
+fun Activity.withPermissions(permission: String, listener: () -> Unit) {
+    if (isPermissionGranted(permission) || requestPermission(permission)) {
+        listener.invoke()
+    }
+}
+
+fun Activity.startLoginActivity() {
+    startActivity(Intent(this, LoginActivity::class.java))
+}
+
+fun Activity.startRegistrationActivity() {
+    startActivity(Intent(this, RegistrationActivity::class.java))
+}
+
+fun Activity.startNewsActivity() {
+    startActivity(Intent(this, NewsActivity::class.java))
+}
```

Comment:
In general, not a good practice.
At least move such navigation methods to separate file, also can have default param Bundle
fun Activity.startLoginActivity(bundle: Bundle = Bundle())